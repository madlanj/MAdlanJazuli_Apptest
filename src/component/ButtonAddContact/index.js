import React from 'react'
import { StyleSheet, TouchableOpacity, Image} from 'react-native'


const ButtonAddContact = ({navigation}) => {
    return (
        <TouchableOpacity
                    style={styles.touchableOpacityStyle}
                    
            >
                <Image style={styles.floatingButtonStyle}
                    source={require('../../image/icon-plus.png')}
                />
            </TouchableOpacity> 

    )
}

export default ButtonAddContact

const styles = StyleSheet.create({
    touchableOpacityStyle: {
        position: 'absolute',
        width: 50,
        height: 50,
        alignItems: 'center',
        justifyContent: 'center',
        right: 10,
        bottom: 10,
    },
    floatingButtonStyle: {
        resizeMode: 'contain',
        width: 50,
        height: 50,
    },
})
