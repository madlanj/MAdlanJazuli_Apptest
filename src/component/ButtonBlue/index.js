import React from 'react'
import { StyleSheet, TouchableOpacity, Text} from 'react-native'


const ButtonBlue = ({textButton, onPress}) => {
    return (
        <TouchableOpacity style={ styles.btnEdit } onPress={ onPress }>
            <Text style={{ color: 'white', fontWeight: 'bold'}}>
                {textButton}
            </Text>
        </TouchableOpacity>  

    )
}

export default ButtonBlue

const styles = StyleSheet.create({
    btnEdit : {
        width:'35%',
        height:35,
        alignItems: 'center',   
        justifyContent: 'center',
        borderRadius:10,
        backgroundColor: "#2b9fe5"
    },
})
