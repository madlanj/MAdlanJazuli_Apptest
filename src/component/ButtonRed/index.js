import React from 'react'
import { StyleSheet, TouchableOpacity, Text} from 'react-native'


const ButtonRed = ({textButton, onPress}) => {
    return (
        <TouchableOpacity style={ styles.btnEdit } onPress={ onPress }>
            <Text style={{ color: 'white', fontWeight: 'bold'}}>
                {textButton}
            </Text>
        </TouchableOpacity>  

    )
}

export default ButtonRed

const styles = StyleSheet.create({
    btnEdit : {
        width:'35%',
        height:35,
        alignItems: 'center',   
        justifyContent: 'center',
        borderRadius:10,
        backgroundColor: "#f03737"
    },
})
