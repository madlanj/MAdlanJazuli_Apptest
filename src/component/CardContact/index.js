import React from 'react'
import { StyleSheet, Text, View, TouchableOpacity, Image } from 'react-native'

const CardContact = ({data, onClick}) => {
    let photo = {uri: data.photo}
    return (
        <TouchableOpacity onPress={(onClick) }>
            <View style={{
                borderRadius:8,
                flexDirection: 'row',
                height: 90,
                borderBottomWidth:1,
                borderColor: "#e1e1e1"
            }}>
                <View style={{ width:'26%',  borderRadius: 8,  alignItems:'center', justifyContent:'center'}}>
                    <Image source={ photo } 
                        style={{ width: 80, height: 80, borderRadius:40, borderWidth:1}} />
                </View>
                <View style={{ 
                marginHorizontal: 14,
                marginVertical: 8,}}>
                    <Text style={{ fontWeight: 'bold', fontSize: 16}}>{data.firstName} {data.lastName}</Text>
                    <Text >{data.age} Years Old</Text>
                </View>
                
            </View>
        </TouchableOpacity>

    )
}

export default CardContact

const styles = StyleSheet.create({})
