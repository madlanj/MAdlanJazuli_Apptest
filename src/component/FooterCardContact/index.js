import React from 'react'
import { StyleSheet, Text, View } from 'react-native'
import { Loading} from '..';
import { useSelector } from 'react-redux';

const FooterCardContact = () => {
    const state = useSelector(state => state.contact);

    return (
        <View style= {{
            position: 'relative',
            marginTop: 10,
            marginBottom: 10,
            paddingBottom: 20
        }}>
            { state.loading ? <Loading textLoading="Please Wait"/> : null }
        </View>
    )
}

export default FooterCardContact

const styles = StyleSheet.create({})
