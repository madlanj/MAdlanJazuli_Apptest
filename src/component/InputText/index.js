import React from 'react'
import { StyleSheet, TextInput, View } from 'react-native'


const InputText = ({value, onChangeText}) => {
    return (
        <TextInput style={styles.textInput} value={value} onChangeText={ onChangeText } />

    )
}

export default InputText

const styles = StyleSheet.create({
    textInput : {
        borderWidth:1, 
        borderColor:'#d4d4d4', 
        borderRadius:10, 
        marginTop:5, 
        height:36, 
        paddingHorizontal:10,        
        alignItems: 'center',   
        justifyContent: 'center',
    },
})
