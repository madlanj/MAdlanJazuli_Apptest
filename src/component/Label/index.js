import React from 'react'
import { StyleSheet, Text, View } from 'react-native'

const Label = ({textLabel}) => {
    return (
            <Text style={styles.label}>{textLabel}</Text>
        
    )
}

export default Label

const styles = StyleSheet.create({
    label : {
        marginTop: 10
    },
})
