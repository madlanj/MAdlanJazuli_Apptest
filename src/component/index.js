import Label from "./Label";
import InputText from "./InputText";
import ButtonBlue from "./ButtonBlue";
import ButtonRed from "./ButtonRed";
import ButtonAddContact from "./ButtonAddContact";
import CardContact from "./CardContact";
import Loading from "./Loading";

export {
    Label,
    InputText,
    ButtonBlue,
    ButtonRed,
    ButtonAddContact,
    CardContact,
    Loading
}