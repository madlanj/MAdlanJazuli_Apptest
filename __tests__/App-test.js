import 'react-native';
import React from 'react';
import { ButtonBlue, ButtonRed, CardContact, InputText, Label, Loading } from '../src/component';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders Button Blue', () => {
  const tree = renderer.create(<ButtonBlue />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Button Red', () => {
  const tree = renderer.create(<ButtonRed />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Card Contact', () => {
  const data = {
      "photo" : "https://sdfs", 
      "firstName" : "bayu",
      "lastName" : "air",
      "age" : 30
  }
  const tree = renderer.create(<CardContact 
    data={data}  />).toJSON();
  expect(tree).toMatchSnapshot();
});


it('renders INput Text', () => {
  const tree = renderer.create(<InputText />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Label', () => {
  const tree = renderer.create(<Label />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Loading', () => {
  const tree = renderer.create(<Loading />).toJSON();
  expect(tree).toMatchSnapshot();
});