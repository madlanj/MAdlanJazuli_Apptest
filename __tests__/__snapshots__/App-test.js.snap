// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button Blue 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2b9fe5",
      "borderRadius": 10,
      "height": 35,
      "justifyContent": "center",
      "opacity": 1,
      "width": "35%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontWeight": "bold",
      }
    }
  />
</View>
`;

exports[`renders Button Red 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f03737",
      "borderRadius": 10,
      "height": 35,
      "justifyContent": "center",
      "opacity": 1,
      "width": "35%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontWeight": "bold",
      }
    }
  />
</View>
`;

exports[`renders Card Contact 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#e1e1e1",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": 90,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 8,
          "justifyContent": "center",
          "width": "26%",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://sdfs",
          }
        }
        style={
          Object {
            "borderRadius": 40,
            "borderWidth": 1,
            "height": 80,
            "width": 80,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "marginHorizontal": 14,
          "marginVertical": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        bayu
         
        air
      </Text>
      <Text>
        30
         Years Old
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders INput Text 1`] = `
<TextInput
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#d4d4d4",
      "borderRadius": 10,
      "borderWidth": 1,
      "height": 36,
      "justifyContent": "center",
      "marginTop": 5,
      "paddingHorizontal": 10,
    }
  }
/>
`;

exports[`renders Label 1`] = `
<Text
  style={
    Object {
      "marginTop": 10,
    }
  }
/>
`;

exports[`renders Loading 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginTop": 10,
    }
  }
>
  <ActivityIndicator
    color="blue"
    size="small"
  />
  <Text>
     
  </Text>
</View>
`;
